pkg/diagnostics/cluster/aggregated_logging/daemonsets.go

```diff
@@ -0,0 +1,109 @@
+package aggregated_logging
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapisext "k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+const daemonSetNoLabeledNodes = `
+There are no nodes that match the selector for DaemonSet '%[1]s'.  
+An example of a command to target a specific node for this DaemonSet:
+
+   oc label node/ip-172-18-2-170.ec2.internal %[2]s
+
+or to label them all:
+
+   oc label node --all %[2]s
+`
+
+const daemonSetPartialNodesLabeled = `
+There are some nodes that match the selector for DaemonSet '%s'.  
+A list of matching nodes can be discovered by running:
+
+   oc get nodes -l %s
+`
+const daemonSetNoPodsFound = `
+There were no pods found that match DaemonSet '%s' with matchLabels '%s'
+`
+const daemonSetPodsNotRunning = `
+The Pod '%[1]s' matched by DaemonSet '%[2]s' is not in '%[3]s' status: %[4]s. 
+
+Depending upon the state, this could mean there is an error running the image 
+for one or more pod containers, the node could be pulling images, etc.  Try running
+the following commands to get additional information:
+
+  oc describe pod %[1]s -n %[5]s
+  oc logs %[1]s -n %[5]s
+  oc get events -n %[5]s
+`
+
+var loggingInfraFluentdSelector = labels.Set{loggingInfraKey: "fluentd"}
+
+func checkDaemonSets(r diagnosticReporter, adapter daemonsetAdapter, project string) {
+	r.Info("AGL0400", fmt.Sprintf("Checking DaemonSets in project '%s'...", project))
+	dsList, err := adapter.daemonsets(project, kapi.ListOptions{LabelSelector: loggingInfraFluentdSelector.AsSelector()})
+	if err != nil {
+		r.Error("AGL0405", err, fmt.Sprintf("There was an error while trying to retrieve the logging DaemonSets in project '%s'", project))
```

Comment:
This would almost certainly be a transient error, would probably be nice to say that and print the error (which would probably be something like "connection reset" or whatever).